lib/CL/pocl_networking.c

```diff
@@ -170,3 +171,18 @@ pocl_remote_client_set_socket_options (int socket_fd, int bufsize, int is_fast)
 
   return 0;
 }
+
+int
+pocl_parse_cid (const char *str, unsigned int *cid)
+{
+  char *endptr;
+  unsigned long t = strtoul (str + 6, &endptr, 10);
```

Comment:
This looks rather dangerous to the uninitiated. Please add some comments to document what is going on.